subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -515,18 +508,23 @@ public boolean isPresent() {
         }
 
         @Override
-        public void configure(final Action<? super Task> action) {
-            configureLater(name, action);
+        public void configure(final Action<? super T> action) {
+            configureEachLater(new Action<Task>() {
```

Comment:
@ghale do you have any ideas how we could push this down into the store maybe? It seems pretty dirty to loop over all the tasks and compare them by name (I'll admit that I added this originally). It might also help the other problem this PR introduces by allowing us to drop the initial configuration block if the task is overwritten.